notifications/pipeline_steps/base.py

```diff
@@ -24,5 +24,7 @@ def process_message(message):
         logging.info("%s has received %s %s", message.current_step, message.name, message.uuid)
         if pytz.utc.localize(datetime.utcnow()) > message.expiration_time:
```

Comment:
If the check happens at L65 in pipeline_manager.py right before calling process_message, the check will happen  right after the message is dequeued and immediately before calling the specific pipeline step's `process_message`.  Once again, we don't want each step to implement its own expiration check.  That should be verified within a single place.